### DISCUSSION TITLE
Hyper-modeling

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#1133] [Q&A] [answered]
now.i use the Hyper-modeling demo by my bim file (dgn to ibim by snopshot),but not success,(the marker not show and the sheet model not show in 3dview),
I don't know the file have problem or   code have  problem!

**Comment by paprophet** [comment]
@pmconne  Can I send you the bim file to your email and  solve the problem！

**Comment by pmconne** [comment]
Use [HyperModeling.isSupportedForIModel](https://github.com/imodeljs/imodeljs/blob/871d57b959138ce6cdfcd6481508a4a2f9100a51/core/hypermodeling/src/HyperModeling.ts#L174). Does it return true or false?

If it returns false, step into that function in the debugger to see why it returned false:
- If it caught an exception then the connector is using an old version of the BisCore schema.
- If it found 0 rows in SectionDrawingLocation then either your .dgn contained no section markers or the connector did not convert them.

If it returns true, show your code.

**Reply by paprophet** [reply]
Use HyperModeling.isSupportedForIModel, and rows=0
the snapshot is update today afternoon，and the dgn have the markers.
![image](https://user-images.githubusercontent.com/18022720/113882946-4e81b280-97f0-11eb-8eb4-c6a0018647b9.png)

**Reply by pmconne** [reply]
Send the .dgn and the .bim to my email please.

**Comment by pmconne** [comment, marked as answer]
Please click the gear icon in iTwin Snapshot, click "iModel Connectors", locate MicroStation, and tell us the version number. For example, my version is 2.1.0.512:
![image](https://user-images.githubusercontent.com/22944042/113889267-10a17180-9791-11eb-8427-4ffafbd198d0.png)

**Reply by paprophet** [reply]
![image](https://user-images.githubusercontent.com/18022720/113893499-abce3180-97f9-11eb-97bc-3226013c3f1f.png)

**Reply by jffmarker** [reply]
To force hypermodeling to be enabled, you can set allow-hypermodeling=1 as an environment variable, and then relaunch the iTwin Snapshot App. @abeesh

**Reply by pmconne** [reply]
@jffmarker that is only a temporary workaround - the app will be updated to enable conversion of section drawing locations, right?

**Reply by jffmarker** [reply]
That's what @abeesh claims, yes. Connectors are on a 2 week release cycle, so you may not see it default to `true` for 2-4 more weeks depending when the change gets taken.

**Reply by paprophet** [reply]
thank you ! @pmconne 
the marker show success ,but there have a error 
i find when i click the 'A-A' marker,then to 2d view and then
 click the [return to 3d view]    show error  
but click the 'C-C' marker ,and not show error

![image](https://user-images.githubusercontent.com/18022720/113967973-c25da280-9864-11eb-95b3-1259c0acea97.png)

![image](https://user-images.githubusercontent.com/18022720/113967995-ca1d4700-9864-11eb-8296-11f507c7cbcf.png)

**Reply by paprophet** [reply]
the problem exist,

**Reply by paprophet** [reply]
@pmconne   do you find the error, 
rotateView the current view ,and change to 'A-A' 2d view, then return to 3d view ,not show this error and success.
so i do not know what viewstate  data can change success? and how catch this error!

**Reply by pmconne** [reply]
I continue to fail to reproduce this issue. If you want me to investigate further, please zip up and email all of your code.
